wdl-analysis/src/eval/v1.rs

```diff
@@ -1100,3 +1109,123 @@ impl<N: TreeNode> Default for CommonAncestorFinder<N> {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use wdl_ast::Document;
+
+    #[test]
+    fn test_input_dependency_handling() {
+        let source = r#"
+        version 1.1
+
+        task my_task {
+            input {
+                Int i
+            }
+
+            command <<<>>>
+
+            output {
+                Int out = i
+            }
+        }
+
+        workflow foo {
+            input {
+                Int x = 10
+                Int y = t1.out
+            }
+
+            call my_task as t1 { input: i = x }
+            call my_task as t2 { input: i = y }
+        }
+        "#;
+
+        let (document, diagnostics) = Document::parse(source);
+        assert!(
+            diagnostics.is_empty(),
+            "parsing should succeed without diagnostics"
+        );
+
+        let workflow = document
+            .ast()
+            .into_v1()
+            .expect("document should be v1")
+            .workflows()
+            .next()
+            .expect("document should have a workflow");
+
+        let mut diagnostics = Vec::new();
+
+        // Testing without providing inputs i.e. static analysis
+        let graph = WorkflowGraphBuilder::default().build(&workflow, &mut diagnostics, |_| false);
+
+        let t1_out = graph
+            .node_indices()
+            .find(|i| {
+                if let WorkflowGraphNode::Call(call) = &graph[*i] {
+                    call.alias().map(|a| a.name().text().to_string()) == Some("t1".to_string())
+                } else {
+                    false
+                }
+            })
+            .expect("t1 node not found");
+
+        let y = graph
+            .node_indices()
+            .find(|i| {
+                if let WorkflowGraphNode::Input(input) = &graph[*i] {
+                    input.name().text().to_string() == "y"
```

Comment:
To make clippy happy:

```suggestion
                    input.name().text() == "y"
```